aws_lambda_powertools/shared/json_encoder.py

```diff
@@ -4,13 +4,27 @@
 
 
 class Encoder(json.JSONEncoder):
-    """
-    Custom JSON encoder to allow for serialization of Decimals, similar to the serializer used by Lambda internally.
+    """Custom JSON encoder to allow for serialization of Decimals, Pydantic and Dataclasses.
+
+    It's similar to the serializer used by Lambda internally.
     """
 
     def default(self, obj):
         if isinstance(obj, decimal.Decimal):
             if obj.is_nan():
                 return math.nan
             return str(obj)
+
+        # Pydantic model (v1/v2)
+        if hasattr(obj, "json"):
```

Comment:
```suggestion
        if hasattr(obj, "root_validators") or hasattr(obj, "__pre_root_validators__"):
```

v2 = `root_validators`
v1 = `__pre_root_validators__`

Maybe this works for v1and v2

Reply:
tested - `json` is available in both v1 and v2, but deprecated in v2. Moved to a standalone function so it can be used elsewhere